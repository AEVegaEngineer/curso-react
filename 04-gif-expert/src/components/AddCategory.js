import React, { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategory = ({setCategories}) => { // esto viene en props pero se desestructura directamente a setCategories

    const [inputValue, setInputValue] = useState('');
    const handleInputChange = (e) => {
        //console.log(e.target.value);
        setInputValue(e.target.value);
        //console.log("handle input change llamado");
    }
    const handleSubmit = (e) => {
        e.preventDefault();

        if( inputValue.trim().length > 2 )
        {
            setCategories( cats => [ inputValue, ...cats]); // se llama directamente el callback del estado anterior 'cats'            
            //props.setCategories([...props.categories, inputValue]) // si se quisiese usar props, habria q obtener tambien las categorias
            setInputValue('');
        }
        
    }
    return (
        <form onSubmit={handleSubmit}>
            <input type='text' value={inputValue} onChange={handleInputChange}/>
        </form>
    );
}
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}