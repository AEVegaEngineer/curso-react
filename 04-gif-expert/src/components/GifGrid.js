import React, { useState, useEffect } from 'react'
import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({category}) => {
    
    const [images, setImages] = useState([]);

    useEffect(() => {
        getGifs(category).then( imgs => setImages(imgs));
    }, [category]);

    

    return (
        <>
            <h3>{category}</h3>
            <div className = "card-grid">
                {
                    images.map( img => {
                        //return <GifGrid gif={gif} key={gif}/>
                        return <GifGridItem key={img.id} {...img}/> // se desestructura la imagen y se envia como parametro al componente
                    })
                }
            </div>
        </>
    )
}

