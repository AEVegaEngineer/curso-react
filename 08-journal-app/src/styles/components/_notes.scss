.notes__main-content{
  display: flex;
  flex-direction: column;
  height: 100%;
}

.notes__appbar{
  align-items: center;
  background-color: $primary;
  color:white;
  display: flex;
  justify-content: space-between;
  padding: 10px 20px 10px 20px;
}

.notes__content{
  display: flex;
  flex-direction: column;
  height: 100%;
  padding: 20px;  
}

.notes__title-input{
  border:none;
  &:focus{
    outline:none;
  }
}

.notes__title-input{
  color: $dark-grey;
  font-size: 25px;
  font-weight: 700;
  margin-bottom: 10px;
}

.notes__textarea{
  // quitando bordes
  border: none !important;
  outline:0px !important;
  -webkit-appearance:none;
  box-shadow: none !important;

  color: dark-grey;
  font-size:20px;
  flex: 1 1 auto; // que se pueda expandir lo mas posible
  resize: none; // quitar el icono del resize en la esquina del textarea
}

.notes__image img{
  box-shadow: 5px 5px $dark-grey;
  height: 150px;
}