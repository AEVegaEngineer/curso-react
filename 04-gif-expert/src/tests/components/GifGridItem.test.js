import React from 'react' 
import { shallow } from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem'


describe('Pruebas en <GifGridItem/>', () => {
    const title = "esto es un titulo";
    const url = "http://localhost/titulo1";
    let wrapper;

    beforeEach( () => {
        wrapper = shallow(<GifGridItem title={title} url={url}/>);
    });

    test('Debe de mostrar <GifGridItem/> correctamente', () => {
        //wrapper = shallow(<GifGridItem title="esto es el ultimo titulo" url="http://localhost/titulo3"/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de tener un parrafo con el title', () => {
         const p = wrapper.find('p');
         expect( p.text().trim() ).toBe(title);
    });

    test('debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        //console.log( img.prop('src') )
        expect( img.prop('src') ).toBe(url);
        expect( img.prop('alt') ).toBe(title);
    });    
    
    test('debe tener animate__fadeInDown', () => {
        const div = wrapper.find('div');
        //animate__fadeInDown
        //console.log(div.prop('className'))
        const className = div.prop('className');
        expect( className.includes('animate__fadeInDown') ).toBe(true);
    });

});

