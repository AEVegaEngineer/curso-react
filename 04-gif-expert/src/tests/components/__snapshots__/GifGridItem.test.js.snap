// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> Debe de mostrar <GifGridItem/> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="esto es un titulo"
    src="http://localhost/titulo1"
  />
  <p>
     
    esto es un titulo
     
  </p>
</div>
`;
